src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx

```diff
@@ -252,20 +253,68 @@ class StacktraceLink extends AsyncComponent<Props, State> {
     return null;
   }
 
-  renderMatchNoUrl() {
+  renderHovercard() {
+    const error = this.match.error;
+    const url = this.match.attemptedUrl;
+    const {frame} = this.props;
     const {config} = this.match;
+    return (
+      <React.Fragment>
+        <StyledHovercard
+          header={
+            error === 'stack_root_mismatch' ? (
+              <span>{t('Mismatch between filename and stack root')}</span>
+            ) : (
+              <span>{t('Unable to find source code url')}</span>
+            )
+          }
+          body={
+            error === 'stack_root_mismatch' ? (
+              <HeaderContainer>
+                <HovercardLine>
+                  {`filename: `} <code>{`${frame.filename}`}</code>
+                </HovercardLine>
+                <HovercardLine>
+                  {`stack root: `} <code>{`${config?.stackRoot}`}</code>
+                </HovercardLine>
+              </HeaderContainer>
+            ) : (
+              <HeaderContainer>
+                <div>{`${url}`}</div>
```

Comment:
@MeredithAnya Nit: you could just have `{url}`, no need for string interpolation